docs/contributor/image_mode.md

```diff
@@ -0,0 +1,217 @@
+# Image Mode for MicroShift
+
+Image mode is a new approach to operating system deployment that lets users build,
+deploy, and manage Red Hat Enterprise Linux as a bootable container (`bootc`) image.
+MicroShift build and deployment procedures can utilize bootable containers to
+benefit from this technology.
+
+This document demonstrates how to:
+* Create a bootable container image and boot a RHEL operating system from this
+image using `podman`
+* Store a `bootc` image in a remote registry and use it for installing a new
+virtual machine
+* Upgrade an existing `rpm-ostree` system created with `composer-cli` tools to
+use `bootc` images
+
+> See [Image mode for Red Hat Enterprise Linux](https://developers.redhat.com/products/rhel-image-mode/overview)
+for more information.
+
+The procedures described below require the following setup:
+* A `physical hypervisor host` with the [libvirt](https://libvirt.org/) virtualization
+platform for starting virtual machines that run RHEL OS containing MicroShift binaries
+* A `development virtual machine` set up according to the [MicroShift Development Environment](./devenv_setup.md)
+instructions for building MicroShift `bootc` images and running containers
+* A `remote registry` (i.e. `quay.io`) for storing and accessing `bootc` images
+
+## Run MicroShift Bootc Image
+
+Log into the `development virtual machine` with the `microshift` user credentials.
+
+### Build Image
+
+Create the `Containerfile` file with the following contents.
+
+```
+FROM registry.redhat.io/rhel9/rhel-bootc:9.4
+
+RUN dnf config-manager \
+        --set-enabled rhocp-4.16-for-rhel-9-x86_64-rpms \
+        --set-enabled fast-datapath-for-rhel-9-x86_64-rpms
+RUN dnf install -y firewalld microshift && \
+    systemctl enable microshift && \
+    dnf clean all
+
+# Create a default 'redhat' user, allowing it to run sudo commands
+ARG USER_PASSWD=\$5\$XDVQ6DxT8S5YWLV7\$8f2om5JfjK56v9ofUkUAwZXTxJl3Sqnc9yPnza4xoJ0
+RUN useradd -m -d /var/home/redhat -G wheel -p ${USER_PASSWD} redhat
+
+# Configure OpenShift pull secret
+ARG PULL_SECRET
+RUN echo $PULL_SECRET > /etc/crio/openshift-pull-secret && \
+    chmod 600 /etc/crio/openshift-pull-secret
```

Comment:
I would expect an `ADD` in such context but I guess for such guide it's easier to pass with --build-arg rather than instruct to copy the file into the build context.

Reply:
Yes, that was the original implementation by copy, but I got comments to make it easier.